### ISSUE TITLE
AzurePsCmdlet error handling for CloudException gives generic messages

### ISSUE THREAD

**Issue by begoldsm** [Azure/azure-powershell#1742]
In powershell, when a services throws a Microsoft.Rest.Azure.CloudException (from the AutoRest world) the message is populated as follows:
"Exception of type 'Microsoft.Rest.Azure.CloudException' was thrown." This is not a particularly useful message for the end user. After debugging, it appears that this is now how the CloudExceptions are thrown by default in AutoRest, with the actual message of the exception in the WebResponse.

This issue tracks fixing CloudException logging/reporting to be the same as it was previously with hyak generated clients that threw Hyak.Common.CloudExceptions.

**Comment by markcowl** [comment]
### Root Cause
Exceptiosn caught by PowerShell instead of cmdlet code

### Proposed Solution
-  Dependent on #1135 
- Once ProcessRecord exceptions are caught, output a meaningful ErrorRecord that conatins the RequestId (and ClientRequestId if available) from the response.

### Cost (Work Units): 2

**Comment by ljt91** [comment]
+1 - Just hit this with New-AzureRmRoleAssignment.

**Comment by markcowl** [comment]
Closing as duplicate of  #3648 and #3649 (which have two concrete solutiosn for this)
